pkg/controller/druid/handler.go

```diff
@@ -328,6 +339,44 @@ func deployDruidCluster(sdk client.Client, m *v1alpha1.Druid) error {
 	return nil
 }
 
+func checkCrashStatus(sdk client.Client, m *v1alpha1.Druid) {
+	podList := podList()
+	listOpts := []client.ListOption{
+		client.InNamespace(m.Namespace),
+	}
+	if err := sdk.List(context.TODO(), podList, listOpts...); err != nil {
+		e := fmt.Errorf("failed to list pods for [%s:%s] due to [%s]", m.Kind, m.Name, err.Error())
+		sendEvent(sdk, m, v1.EventTypeWarning, "LIST_FAIL", e.Error())
+		logger.Error(e, e.Error(), "name", m.Name, "namespace", m.Namespace)
+	}
+
+	pod := make([]*v1.Pod, 0)
+	for i := range podList.Items {
+		pod = append(pod, &podList.Items[i])
+	}
+
+	for _, p := range pod {
+		if p.Status.ContainerStatuses[0].RestartCount > 1 {
+			for _, condition := range p.Status.Conditions {
+				if condition.Type == "Ready" {
+					if p.Status.Phase != "Running" || condition.Status == "False" {
```

Comment:
can you add a inline code comment around here, explaining the logic around the if checks

Reply:
sure